### ISSUE TITLE
Integrate "ScriptTask" plugin into MeshCentral core

### ISSUE THREAD

**Issue by theoneandonly-vector** [Ylianst/MeshCentral#3613]
"This is a full computer management web site."

to make this statement true, I think this plugin should be integrated / supported in the main software:
https://github.com/ryanblenis/MeshCentral-ScriptTask

**Comment by PathfinderNetworks** [comment]
That would certainly be a nice feature.  Me, I don't run any plugins as I don't want to have to troubleshoot any issues they may cause (especially seeing as how I keep my MeshCentral install always updated with the most current version).

**Comment by Ylianst** [comment]
Fixed poor (low quality) title. I am certainly not going to be doing anything because a statement is true or not.

I have done some early work to integrate that plugin into MeshCentral, but I need to do a lot more work before releasing it.

**Comment by silversword411** [comment]
If you want scripting maybe you should look at an rmm?

https://github.com/wh1te909/tacticalrmm

**Comment by theoneandonly-vector** [comment]
> If you want scripting maybe you should look at an rmm?
> 
> https://github.com/wh1te909/tacticalrmm

this look svery nice, but I would like to keep a central setup for all types of clients.
also SSO is needed.

**Comment by cgb** [comment]
Just a heads up about tacticalrmm.  I'd been monitoring it for a while but this event put me off:
https://www.reddit.com/r/sysadmin/comments/rq5g9h/a_few_concerns_with_tacticalrmm_am_i_being/

There is an explanation here:
https://www.reddit.com/r/msp/comments/rqm0go/a_statement_from_the_founder_of_tacticalrmm/

But it's a major red flag.  They have the rmmagent on github but are only making binary releases while the last published code is from Apr 2021.

Tread carefully.

**Comment by Ylianst** [comment]
Oh wow! I have not been following TacticalRMM much, apart from knowing they using MeshCentral. This is all news to me.

Getting a crypto minor in a release is really bad (!!!). Also, I am not sure how you can make releases without having the source on GitHub first unless your doing it on purpose. MeshCentral, MeshAgent and all related projects are always checked into GitHub before release. We have only one repo and it's the public one, everything we do, typo's and all are public.

**Comment by silversword411** [comment]
There was never a crypto miner in any Tactical RMM release.

**Comment by silversword411** [comment]
@Ylianst 

TLDR
- main dev had a ftp site at tacticalrmm.io (was an old personal ftp server that got renamed for tacticalrmm use) for downloading python install files for windows agent installs if github was down at time of agent installation (because that seems to be a sporadic bi-monthly occurrence).
- On that same ftp site there were other personal agent installers that he used himself. Found (I think it was) golangs ability to bundle other apps and was playing with it for his own use and putting a personal monero miner in his personal custom agent builds/installers.
- Because it was at tacticalrmm.io everyone assumed TRMM was installing those agents. Never did. 
- People that didn't read his reddit looong response and didn't pay attention to details see monero and trmm and #HairOnFire

There was **NEVER** a monero miner in any part of Tactical RMM's official repo, any of it's linked resources, or install processes.

**Comment by petervanv** [comment]
He says it self in the statemetn, there was an miner !!

![image](https://user-images.githubusercontent.com/58996467/154519791-1a1851e8-1f99-4893-a4e0-03fe95a8839e.png)

**Comment by silversword411** [comment]
> He says it self in the statemetn, there was an miner !!

OMG. Read....the...**ACTUAL**...words...of...the...statement...in....it's...entirety. You just screenshot it and posted it for gods sake! Details really do matter.

The monero miner was in the file `winagent-v1.98.61.exe`

That file, it's contents, AND it's naming format as **NOTHING**, **ZERO**, **ZIP**, **NADDA** to do with the installation/updating of TacticalRMM, it's sources, update process or **ANY** part of the installation and distribution used by all Tactical RMM users.

------------------------------

That file and it's naming format is only created by an already installed Tactical RMM server and personally creating an agent installation .exe for your personal workstation deployment process. It is not part of any source used by anyone installing Tactical RMM. 

1) He installed Tactical on his own private server. 
2) He then used the agent source, and built a custom agent installer on his own machine in vscode and put his personal monero agent into an agent installer that will only connect to **HIS TRMM** server and will not even **run** unless he approves that monero miner to activate itself. 

His mistake...was taking that personal and private agent installer...and uploaded it to the ftp server located at `files.tacticalrmm.io`....where noone would ever have found it unless you:
- Looked at github source for the agent
- saw the backup url for downloading the `py38-x32.zip` and `py38-x64.zip`
- Opened `https://files.tacticalrmm.io/` and started looking at what OTHER files might be there. 
- Found his personal agent installer
- Manually downloaded it and ran that .exe not knowing what it was.

----------------------

A meshcentral issue chain is not the place to be rehashing this discussion. Look me up on the Tactical discord if you still can't understand something.

**Comment by PathfinderNetworks** [comment]
For me at least, having something like this (a mining app, etc) even remotely related to/near something like an RMM tool is enough for me to steer clear of it.  Just no.  The integrity of my company and my customers security is FAR to important to even take a tiny, remote chance of something like this finding its way on to my server/their systems.

**Comment by cgb** [comment]
Hey @Ylianst apologies for bringing the tacticalrmm controversy into this issue, I intentionally didn't pass judgement in my earlier comment and just linked to the public discourse to let others form a view. I just got concerned when there is a recommendation for a related "open source" RMM and your users weren't aware of the incident..

**Comment by si458** [comment]
bad bot, not stale

**Comment by ne0YT** [comment]
*bad bot, not stale

**Comment by ne0YT** [comment]
*bad bot, not stale

**Comment by ne0YT** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

*bad bot, not stale

**Comment by panreyes** [comment]
But... is it stale or isn't it? :)
@si458 @ryanblenis Are you actively working on embedding/integrating ScriptTask into MeshCentral?

In my opinion, I think I currently prefer to have it as a plugin, but as you think it's best.
The only thing I miss is having it as a tab, instead of being a subtab in plugins. We already have good wide monitors everyone, so it shouldn't be a problem in mosts cases.

**Comment by ryanblenis** [comment]
@panreyes Ylian took on that task and asked me if it was ok. I gave him the green light to go ahead but I don't think it made it very far, for better or worse.

I had mixed feelings about it because then any design/functionality/compatibility was essentially off my table (so less for me to do, but less say I had in it) - so good and bad. But either way I was watching to see which way it went.
The best part about plugins is that you can make it do what you want to do with it and didn't need to wait on a merge to get things done, and that devs around the world could keep adding and (in theory) could exponentially grow the abilities of MC without bloating the main project.
Thus I introduced the concept of "first party" and "third party" plugins where some core plugins could be recognized as stable/useful/trusted, and all the other ones people came to find over time.
